modules/vaos/spec/services/eps/base_service_spec.rb

```diff
@@ -24,4 +28,65 @@
       expect(service.config).to be_instance_of(Eps::Configuration)
     end
   end
+
+  describe '#token' do
+    context 'when cache is empty' do
+      before do
+        allow(Rails.cache).to receive(:fetch)
+          .with(described_class::REDIS_TOKEN_KEY, expires_in: described_class::REDIS_TOKEN_TTL)
+          .and_yield
+        allow(service).to receive(:get_token).and_return(mock_token_response)
+      end
+
+      it 'fetches and caches new token' do
+        expect(service.token).to eq('mock_token')
+      end
+    end
+
+    context 'when cache exists' do
+      before do
+        allow(Rails.cache).to receive(:fetch)
+          .with(described_class::REDIS_TOKEN_KEY, expires_in: described_class::REDIS_TOKEN_TTL)
+          .and_return('cached_token')
+      end
+
+      it 'returns cached token' do
+        expect(service.send(:token)).to eq('cached_token')
+      end
+    end
+  end
+
+  describe '#get_token' do
+    let(:config) { instance_double(Eps::Configuration) }
+    let(:jwt_wrapper) { instance_double(Eps::JwtWrapper) }
+
+    before do
+      allow(Eps::Configuration).to receive(:instance).and_return(config)
+      allow(config).to receive_messages(access_token_url: 'http://test.url', grant_type: 'client_credentials',
+                                        scopes: 'test.scope', client_assertion_type: 'urn:test')
+      allow(Eps::JwtWrapper).to receive(:new).and_return(jwt_wrapper)
+      allow(jwt_wrapper).to receive(:sign_assertion).and_return('signed_jwt')
+
+      allow(service).to receive(:perform).and_return(mock_token_response)
+      allow(service).to receive(:with_monitoring).and_yield
+    end
+
+    it 'makes POST request with correct parameters' do
```

Comment:
Can we also add a failure case where the token endpoint doesn't return any access_token?

Reply:
added in 473ac8c3657782d4152056fea4665b548f4db355